### ISSUE TITLE
Add inner_hits support to the collapse field in NativeSearchQuery [DATAES-939]

### ISSUE THREAD

**Issue by spring-projects-issues** [spring-projects/spring-data-elasticsearch#1493]
**[sothawo](https://jira.spring.io/secure/ViewProfile.jspa?name=sothawo)** opened **[DATAES-939](https://jira.spring.io/browse/DATAES-939?redirect=false)** and commented

Currently NativeSearchQuery supports a collapse field by accepting a String. This should be extended to also allow to define inner_hits and other parameters (https://www.elastic.co/guide/en/elasticsearch/reference/7.9/collapse-search-results.html#expand-collapse-results)

---

**Reference URL:** https://stackoverflow.com/questions/64106687/spring-data-elasticsearch-is-inner-hit-supported-at-root-level-on-query

**Comment by saidkilicaslan** [comment]
Hi, is there any workaround to get inner hits with collapse field for now?

**Comment by sothawo** [comment]
NO, this is still an open issue.

**Comment by saidkilicaslan** [comment]
I did the workaround.
I'm using [JestElasticsearchTemplate](https://github.com/VanRoy/spring-data-jest/blob/main/spring-data-jest/src/main/java/com/github/vanroy/springdata/jest/JestElasticsearchTemplate.java) as client and I copied NativeQueryBuilder with an extra method:

` public ExtendedNativeQueryBuilder withCollapse(CollapseBuilder collapseBuilder){
        this.collapseBuilder = collapseBuilder;
        return this;
    }`

and using the client with the new method.

`((JestElasticsearchTemplate) elasticsearchTemplate).query(new ExtendedNativeQueryBuilder()
            .withIndices("indice").withFilter(QueryBuilders.matchAllQuery())
            .withCollapse(new CollapseBuilder("collapse_field").setInnerHits(new InnerHitBuilder("inner_hit_name")))
            .build(), extractor);`

as a query result, I'm able to retrieve innert_hits inside the JSON object.
